### PR TITLE
Docs: api keys

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

Adds planning references for API keys.

**Why do we need this feature?**

Docs revamp work

**Who is this feature for?**

It's a part of a larger authNZ doc refactor.

### CODE REVIEW

File: docs/sources/auth/planning/index.md [hunk 1]
```diff
@@ -122,6 +122,10 @@ Service account access tokens inherit permissions from the service account.
 
 ### 🚧 API keys
 
+Grafana API keys can be used to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].
```
Comment (chri2547) [inline docs/sources/auth/planning/index.md:125]:
```suggestion
You can use Grafana API keys to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].
```

Comment (chri2547) [inline docs/sources/auth/planning/index.md:125]: Should we point users to the service account docs as the way to do this? Eric has made some nice updates to those docs recently (not sure if merged in yet).

File: docs/sources/auth/planning/index.md [hunk 2]
```diff
@@ -122,6 +122,10 @@ Service account access tokens inherit permissions from the service account.
...
+
+> **Note:** If you use Grafana v8.5 or newer, use service accounts instead of API keys. For more information, refer to [Grafana service accounts]({{< relref "../service-accounts/" >}}).
```
Comment (chri2547) [inline docs/sources/auth/planning/index.md:127]: oh nvm, you do point to the service account docs. :)

File: docs/sources/auth/planning/index.md (hunk 1 above)
Comment (chri2547) [inline docs/sources/auth/planning/index.md:125]: I'm not sure what "to resources with the help of roles" means.

**Review by chri2547 — COMMENTED** [review]
Thanks! Added comments and suggestion.

File: docs/sources/auth/planning/index.md (hunk 1 above)
Comment (linoman) [reply docs/sources/auth/planning/index.md:125]: I meant a limited scope or limited access to resources (editing dashboards, viewing sensitive information) with a well-defined role/permission definition.

If this is not clear at first glance, I should reword it.

Comment (IevaVasiljeva) [reply docs/sources/auth/planning/index.md:125]: You can only use the basic roles (Viewer, Editor, Admin) for API keys. Service accounts are more flexible and Enterprise users can assign specific RBAC roles to them.

File: docs/sources/auth/planning/index.md (hunk 2 above)
Comment (IevaVasiljeva) [reply docs/sources/auth/planning/index.md:127]: I actually think that we should push this info higher (maybe even in the first sentence?) and be very explicit that API keys are deprecated in favour of service accounts.

File: docs/sources/auth/planning/index.md (hunk 1 above)
Comment (linoman) [reply docs/sources/auth/planning/index.md]: Changed!

File: docs/sources/auth/planning/index.md (hunk 2 above)
Comment (linoman) [reply docs/sources/auth/planning/index.md]: I moved and rephrased the warning to the top level.

File: docs/sources/auth/planning/index.md [hunk 3]
```diff
@@ -120,7 +120,11 @@ Service account access tokens inherit permissions from the service account.
 
 ### 🚧 Personal access tokens
 
-### 🚧 API keys
+### API keys
+
+> **Note:** If you use Grafana v8.5 or newer, you should use service accounts instead of API keys. API keys will be deprecated in the near future. For more information, refer to [Grafana service accounts]({{< relref "../service-accounts/" >}}).
+
+You can use Grafana API keys to interact with data sources via HTTP APIs. API keys can have a well-defined and limited scope to resources with the help of [Roles].
```
Comment (IevaVasiljeva) [inline docs/sources/auth/planning/index.md:127]: API keys aren't limited to data sources - they can be used to interact with any Grafana resource.

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**